code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/models/AEPPushTemplate.kt

```diff
@@ -0,0 +1,399 @@
+            Log.debug(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Exception in converting notification badge count to int - ${e.localizedMessage}."
+            )
+        }
+        notificationPriority = NotificationPriority.from(
+            data[PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY]
+        )
+        notificationImportance =
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) getNotificationImportanceFromString(
+                data[PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY]
+            ) else null
+
+        notificationVisibility = getNotificationVisibilityFromString(
+            data[PushTemplateConstants.PushPayloadKeys.NOTIFICATION_VISIBILITY]
+        )
+        channelId =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.CHANNEL_ID, null)
+        templateType =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE, null)
+                ?.let { PushTemplateType.fromString(it) }
+        tag = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TAG, null)
+        val stickyValue =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.STICKY, null)
+        isNotificationSticky =
+            if (stickyValue.isNullOrEmpty()) false else java.lang.Boolean.parseBoolean(
+                stickyValue
+            )
+        ticker = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TICKER, null)
+        expandedBodyText = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT, null
+        )
+        expandedBodyTextColor = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, null
+        )
+        titleTextColor = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.TITLE_TEXT_COLOR, null
+        )
+        smallIconColor = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.SMALL_ICON_COLOR, null
+        )
+        notificationBackgroundColor = DataReader.optString(
+            data,
+            PushTemplateConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR,
+            null
+        )
+        isFromIntent = false
+    }
+
+    /**
+     * Constructor to create a push template object from an [Intent].
+     *
+     * @param intent [Intent] containing key value pairs extracted from a push template data payload
+     */
+    constructor(intent: Intent) {
+        val intentExtras =
+            intent.extras ?: throw IllegalArgumentException("Intent extras are null.")
+        // required values
+        title = intentExtras.getString(PushTemplateConstants.IntentKeys.TITLE_TEXT)
+            ?: throw IllegalArgumentException("Required field \"adb_title\" not found.")
+        body = intentExtras.getString(PushTemplateConstants.IntentKeys.BODY_TEXT)
```

Comment:
Why is there inconsistency between the implementations of the two constructors?
For instance, the other constructor throws an exception when encountering an empty string, while also attempting to read ACC_PAYLOAD_BODY:

```
var bodyText = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.BODY, null)
        if (bodyText.isNullOrEmpty()) {
            bodyText =
                DataReader.optString(
                    data,
                    PushTemplateConstants.PushPayloadKeys.ACC_PAYLOAD_BODY, null
                )
        }
```

Reply:
If this is expected, can you add a comment?

Reply:
The map data is received from the FCM message. My guess is it can have body text as either the `adb_body` or `_msg`.  But in the latter case, it should be converted in the caller extension to the expected adb format. I will leave this one open to confirm with @rymorale.
The intent is from the pending intent constructed by the notification builder classes so its guaranteed to have the key as `bodyText`

Reply:
Verified with Ryan, the caller extension (ACC) needs to covert all the legacy keys to `adb` prefixed one. Removing the logic